spline-shadow/rest-core/src/main/scala/za/co/absa/spline/gateway/rest/model/ExecutionInfo.scala

```diff
@@ -16,18 +16,19 @@
 
 package za.co.absa.spline.gateway.rest.model
 
-import java.util.{Date, UUID}
+import java.util.UUID
 
 import za.co.absa.spline.gateway.rest.model.ExecutionInfo.Id
 
 case class ExecutionInfo
 (
   _id: Id,
-  nativeId: Option[String],
-  startTime: Option[Date],
-  finishTime: Option[Date]
+  _type: String,
+  startTime: Option[Long],
+  endTime: Option[Long],
+  extra: Map[String, Any]
 ) {
-  require(Seq(startTime, finishTime).exists(_.isDefined), "at least one of 'start' or 'finish' time should be defined")
+  def this() = this(null, null, null, null, null)
```

Comment:
I am not familiar with scala but is it necessary to define such a constructor ?

Reply:
It is probably needed for the json serialization

Reply:
Yes, unfortunately it's one of those scala-java edge stones. Arango Java driver require model classes to have a default constructor (the one with no args), that is one of fundamental requirement in Java bean spec. But Scala case classes doesn't follow Java bean spec and do not generate no-arg constructors by default.
I have a plan to write a macro activated by a static annotation that would generate default constructor for an annotated case class. But that's not an easy task, so I saved it for later.